diff --git a/include/grpcpp/impl/codegen/proto_utils.h b/include/grpcpp/impl/codegen/proto_utils.h	
index 2e102135a3..ea68fc7068 100644	
--- a/include/grpcpp/impl/codegen/proto_utils.h	
+++ b/include/grpcpp/impl/codegen/proto_utils.h	
@@ -49,7 +49,7 @@ Status GenericSerialize(const grpc::protobuf::MessageLite& msg, ByteBuffer* bb, 	
                 "ProtoBufferWriter must be a subclass of "	
                 "::protobuf::io::ZeroCopyOutputStream");	
   *own_buffer = true;	
-  int byte_size = msg.ByteSize();	
+  size_t byte_size = msg.ByteSizeLong();	
   if ((size_t)byte_size <= GRPC_SLICE_INLINED_SIZE) {	
     Slice slice(byte_size);	
     // We serialize directly into the allocated slices memory	
@@ -60,7 +60,7 @@ Status GenericSerialize(const grpc::protobuf::MessageLite& msg, ByteBuffer* bb, 	

     return g_core_codegen_interface->ok();	
   }	
-  ProtoBufferWriter writer(bb, kProtoBufferWriterMaxBufferLength, byte_size);	
+  ProtoBufferWriter writer(bb, kProtoBufferWriterMaxBufferLength, (int)byte_size);	
   return msg.SerializeToZeroCopyStream(&writer)	
              ? g_core_codegen_interface->ok()	
              : Status(StatusCode::INTERNAL, "Failed to serialize message");